airflow/cli/commands/standalone_command.py

```diff
@@ -159,14 +159,28 @@ def calculate_env(self):
         # Make sure we're using a local executor flavour
         executor_class, _ = ExecutorLoader.import_default_executor_cls()
         if not executor_class.is_local:
-            if "sqlite" in conf.get("database", "sql_alchemy_conn"):
-                self.print_output("standalone", "Forcing executor to SequentialExecutor")
-                env["AIRFLOW__CORE__EXECUTOR"] = executor_constants.SEQUENTIAL_EXECUTOR
-            else:
-                self.print_output("standalone", "Forcing executor to LocalExecutor")
-                env["AIRFLOW__CORE__EXECUTOR"] = executor_constants.LOCAL_EXECUTOR
+            env["AIRFLOW__CORE__EXECUTOR"] = self.get_local_executor(
+                database=conf.get("database", "sql_alchemy_conn")
+            )
+            self.print_output("standalone", f"Forcing executor to {env['AIRFLOW__CORE__EXECUTOR']}")
```

Comment:
The more I reflect on this, the more I think the original code is actually fine. This logic is only executed if the executor selected by the user (or provided by the user via plugin) is not local, in which case we select a working default for them. However, they are completely able to implement/select a compatible local executor themselves if they wish to be opinionated about which executor to use.

I think this logic and the helper method below actually might complicate things a bit more than it helps. WDYT?

Also CC: @pierrejeambrun

Reply:
@o-nikolas  The only benefit I see with this code we still remove the need to change this part of the code if someone implements a new local executor, right?

Reply:
>  The only benefit I see with this code we still remove the need to change this part of the code if someone implements a new local executor, right?

If someone develops a new local executor, they are able to use it just fine and they do not need to touch this code. This code is only run if the user fails to use a compatible executor for the standalone setup (`if not executor_class.is_local:`), in which case we choose a workable default for them. I don't think that this defaulting code needs to be pluggable or updated if new executors are added. Because if the user wishes to use any specific (and compatible) executor with standalone they're welcome to do that, and it's as easy as just configuring to use it in airflow.cfg or via `AIRFLOW__CORE__EXECUTOR`. And if that executor they pick is compatible (i.e. passes the `is_local` check) then this code is not run.

It is basically another example of coupling type 7 in the original AIP, here is the snippet:

![Screenshot from 2023-01-20 10-42-49](https://user-images.githubusercontent.com/65743084/213780548-8e4c8514-4bae-4ae7-b550-1b2122042a07.png)

Hopefully that makes sense, it's a bit of a nuanced issue. It is my bad for mis-identifing this piece of code as a point of executor coupling :pray:

Reply:
@o-nikolas Thanks for the context, this makes sense now. I'll revoke these changes.